src/storage-controller/src/lib.rs

```diff
@@ -516,10 +516,66 @@ where
             .get_mut(&instance_id)
             .unwrap_or_else(|| panic!("instance {instance_id} does not exist"));
 
+        let status_now = mz_ore::now::to_datetime((self.now)());
+        let mut source_status_updates = vec![];
+        let mut sink_status_updates = vec![];
+
+        let make_update = |id, object_type| StatusUpdate {
+            id,
+            status: Status::Paused,
+            timestamp: status_now,
+            error: None,
+            hints: BTreeSet::from([format!(
+                "The replica running this {object_type} has been dropped"
+            )]),
+            namespaced_errors: Default::default(),
+            replica_id: Some(replica_id),
+        };
+
         for id in instance.active_ingestions() {
+            let ingestion = self
+                .collections
+                .get_mut(id)
+                .expect("instance contains unknown ingestion");
```

Comment:
Are these two states guaranteed to be updated transactionally? (A log and `continue` here seems less scary if we're not totally sure.)

Reply:
Not transactionally, but these parts can't have concurrency (at least today). We've had this `expect()` here for a long time, so I think it's okay. But I can change this and the other one to a `soft_assert_or_log` as a follow-up, if you like!

Reply:
Mmm... f this is just code movement, I think it's AOK to leave it alone.